PotreeConverterPartitioned/potreeconverterpartitioned.py

```diff
@@ -0,0 +1,533 @@
+                if not copiedBatchNums: # if no batch is waiting to be processed
+                    spaceUtilization = lazBatch["size"] * (self.lazCompressionRatio + self.inputToOutputSizeRatio + self.inputToTmpInputSizeRatio)
+                elif len(copiedBatchNums) == 1: # if a batch is already being indexed
+                    currBatchSize = self.indexingBatchCopier.gettotalSize()
+                    #Assuming that the currBatch is immediately removed before distributiion phase of nextBatch get start
+                    spaceUtilization = (currBatchSize * self.inputToOutputSizeRatio) + (lazBatch["size"] * self.inputToOutputSizeRatio) + (lazBatch["size"] * self.lazCompressionRatio)
+                else: #if a batch is already being indexed and others are waiting
+                    currBatchSize = self.indexingBatchCopier.getBatchSize(heapq.nsmallest(2, copiedBatchNums)[0])
+                    nextBatchSize = self.indexingBatchCopier.getBatchSize(heapq.nsmallest(2, copiedBatchNums)[1])
+                    #Assuming that the currBatch is immediately removed before distributiion phase of nextBatch get start
+                    spaceUtilization = (currBatchSize * self.inputToOutputSizeRatio) + (nextBatchSize * (self.lazCompressionRatio + self.inputToOutputSizeRatio)) + lazBatch["size"] + self.indexingBatchCopier.gettotalSize()
+
+                if spaceUtilization <= self.maxTmpSpaceAvailable:
+                    batchToCopy = lazBatch.copy()
+                    self.lazFilestoProcess.remove(lazBatch)
+                    self.distribution(batchToCopy["files"], batchToCopy["id"])
+                    self.copyBatch([], batchToCopy["size"], self.tmpInputDir, self.indexingBatchCopier,
+                                   self.tmpOutputDir + "", "indexing", batchToCopy["id"])
+                self.testBatchDone(self.distributionBatchCopier, self.tmpOutputDir + "/distribution_done_signals", "distribution")
+                self.testBatchDone(self.indexingBatchCopier, self.tmpOutputDir + "/indexing_done_signals", "indexing")
+
+        while (not self.indexingBatchCopier.isbatchDictEmpty()) or (not self.distributionBatchCopier.isbatchDictEmpty()):
+            if not self.distributionBatchCopier.isbatchDictEmpty():
+                self.testBatchDone(self.distributionBatchCopier, self.tmpOutputDir + "/distribution_done_signals", "distribution")
+            if not self.indexingBatchCopier.isbatchDictEmpty():
+                self.testBatchDone(self.indexingBatchCopier, self.tmpOutputDir + "/indexing_done_signals", "indexing")
+            time.sleep(60)
+        LoggingWrapper.info("Indexing finished. Total " + str(
+            self.numFilesDone["indexing"]) + " files indexed in " + str(
+            len(self.batchesDone["indexing"])) + " partitions. Total " + str(totalBatches) + " partitions, " + str(
+            len(skipPartitions)) +  " skipped" , color="green", bold=True)
+
+
+    def parseBladnr(self, bladnr):
```

Comment:
Given that this should be modified for different naming conventions of input point cloud laz files, moving it to a separate module and importing it would make more sense, and allow users to find it more easily if/when they need to modify it.